tests/tests.py

```diff
@@ -73,15 +78,18 @@ def test_login_returns_serialized_token(self):
         self.assertNotIn(username_field, response.data)
 
     def test_login_returns_serialized_token_and_username_field(self):
-        self.assertEqual(AuthToken.objects.count(), 0)
-        url = reverse('knox_login')
-        self.client.credentials(
-            HTTP_AUTHORIZATION=get_basic_auth_header(self.username, self.password)
-        )
-        knox_settings.USER_SERIALIZER = UserSerializer
-        response = self.client.post(url, {}, format='json')
+
+        with override_settings(REST_KNOX=user_serializer_knox):
+            reload_module(views)
+            self.assertEqual(AuthToken.objects.count(), 0)
+            url = reverse('knox_login')
+            self.client.credentials(
+                HTTP_AUTHORIZATION=get_basic_auth_header(self.username, self.password)
+            )
+            response = self.client.post(url, {}, format='json')
+            self.assertEqual(user_serializer_knox["USER_SERIALIZER"], UserSerializer)
+        reload_module(views)
         self.assertEqual(response.status_code, 200)
-        self.assertNotEqual(knox_settings.USER_SERIALIZER, None)
```

Comment:
I think this should stay in place.

Reply:
If this was to stay it should be moved within the override